plugins/woocommerce/src/Internal/Admin/ProductForm/Form.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * WooCommerce Product Form
+ *
+ * @package Woocommerce ProductForm
+ */
+
+namespace Automattic\WooCommerce\Internal\Admin\ProductForm;
+
+/**
+ * Contains logic for the WooCommerce Product Form.
+ */
+class Form {
```

Comment:
Will there be a separate `VariationForm` class for variations?  Just thinking out loud for the sake of this logic and reducing ambiguity in this class name if necessary.

Reply:
My thought was to actually only keep `Form` and use that class to register new templates, tabs, sections, and fields. I didn't really want to make a standard were we create a new `Form` class for every type of `Form`.

Sounds like Variations might actually be the only exception to this approach which we may of not fully thought through yet. 
I guess I have two thoughts around this:
- Do we want to have the Variation form be defined as a custom template, not available for selection (product creation) but only used to render the form layout for Variations?
- Or do we want to create a new `VariationForm` that mostly inherits from `Form` but is for Variations only.

I would lean towards the first approach more, as we want to treat variations largely the same.